packages/execution-environments/src/common/commands.ts

```diff
@@ -13,20 +17,62 @@ export type CommandMethodsMapping = {
   snapRpc: SnapRpc;
 };
 
+// TODO: Add validation in cases.
+/**
+ * Formats the arguments for the given handler.
+ *
+ * @param origin - The origin of the request.
+ * @param handler - The handler to pass the request to.
+ * @param request - The request object.
+ * @returns The formatted arguments.
+ */
+function getHandlerArguments(
+  origin: Origin,
+  handler: HandlerType,
+  request: JsonRpcRequest<unknown[] | { [key: string]: unknown }>,
+): InvokeSnapArgs {
+  switch (handler) {
+    case HandlerType.OnTransaction: {
+      const { transaction, chainId } = request.params as Record<string, any>;
+      return {
+        origin,
+        transaction,
+        chainId,
+      };
+    }
+
+    case HandlerType.OnRpcRequest:
+      return { origin, request };
+
+    default:
+      return assertExhaustive(handler);
+  }
+}
```

Comment:
For future reference: Is there any reason not to have validation in these cases?

Reply:
What do you mean by validation? At the `snapRpc` level we do a `isHandler` check before passing on the handler to `getHandlerArguments`.